### ISSUE TITLE
When Copying/Publishing Pages, I Get an Exception (AggregateException, Length Parameter Out of Range)

### ISSUE THREAD

**Issue by Nicholas-Westby** [umbraco/Umbraco-CMS#15526]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.0.3

### Bug summary

When I try to copy some nodes or I try to publish a page, I get an exception. I was able to copy and publish pages at first, until I got to about 7 pages. Each page has about 450 widgets (via block list).

### Specifics

I was trying to troubleshoot an issue with Umbraco being slow on publish, so I thought I'd start by reproducing it with a clean Umbraco install using the latest version (just to see if it was because of plugins or custom code). Before I got a chance to finish my testing, I ran into an issue where copying pages was throwing an exception. Then I tried publishing individual pages and it seemed to throw the same exception. While the log left no trace of the error (there are some warnings you can see), I did copy this stack trace from the UI:

```
Received an error from the server
An error occurred
One or more errors occurred. (Exceptions were thrown by listed actions. (Specified argument was out of the range of valid values. (Parameter 'length')))

Exception Details
System.AggregateException, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e: One or more errors occurred. (Exceptions were thrown by listed actions. (Specified argument was out of the range of valid values. (Parameter 'length')))
Stacktrace
at Umbraco.Cms.Infrastructure.Scoping.Scope.TryFinally(Action[] actions)
   at Umbraco.Cms.Infrastructure.Scoping.Scope.Dispose()
   at Umbraco.Cms.Core.Services.ContentService.SaveAndPublish(IContent content, String culture, Int32 userId)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.PublishInternal(ContentItemSave contentItem, String defaultCulture, String cultureForInvariantErrors, Boolean& wasCancelled, String[]& successfulCultures)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.PostSaveInternal[TVariant](ContentItemSave contentItem, Func`3 saveMethod, Func`2 mapToDisplay)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.PostSave(ContentItemSave contentItem)
   at lambda_method6439(Closure, Object)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|26_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
```

Side note: the many warnings you'll see if you review the log match the warning Dan saw back in Umbraco 10: https://twitter.com/DanDiplo/status/1596090169609838597

That warning says the JSON couldn't be parsed despite the JSON it mentions looking fine. No idea if that's related to the main error from this issue, but is certainly something to look into that shouldn't happen.

### Steps to reproduce

Download this ZIP file:

[Umbraco13SpeedTest.zip](https://github.com/umbraco/Umbraco-CMS/files/13824718/Umbraco13SpeedTest.zip)

It contains the Visual Studio solution and the database backup you need to reproduce. Do the following:

* Unzip the file.
* Restore the database (I'm using SQL Server 2022 Express, so that or newer should work).
* Update the connection string to point to your restored database (the server name may have changed).
* Debug the project from Visual Studio 2022 (my specific version is 17.8.3). I debug with Chrome (probably not important though).
* Log in to Umbraco (there is no frontend to this site). The username is "admin@admin.admin" and the password is "adminadmin".
* Publish any page.
* You should see an exception dialog appear that looks like the one below.

<img width="962" alt="no-more-publishing" src="https://github.com/umbraco/Umbraco-CMS/assets/5933222/af9f3250-e3b0-485b-bf28-e3c99d52721a">

To reproduce this from scratch rather than downloading the ZIP, do the following:

* Install the 13.0.3 template using the standard `dotnet new` approach.
* Create a new project in Visual Studio based on that.
* Run through the installer with the standard stuff, making sure to choose settings appropriate for SQL Server 2022.
* Create a block list accepting 3 link types (for content, media, and external links).
* Create a block list accepting 2 widget types (for rich text widget and callout widget).
* Create a typical page content type with a property for widgets.
* Create a typical page with about 450 widgets (I created a few and then used the copy/paste feature to create a bunch exponentially). I filled in the callout widget with 3 links (one for each type), so there'd be about 1000 or so links on that one page (within the callout widgets).
* Copy the page a few times, moving it underneath itself, creating a tree structure with about 7 pages, using a variety of copy approaches (e.g., one at a time, move, copy several at a time, and so on).
* You should see the exception at some point.

### Expected result / actual result

I should be able to create pretty much any number of widgets and pages without an error happening. I should also be able to see that error in the error log rather than have to copy it from the UI. Even if I did have an error, it should be clear about why I am running into the error rather than being very obscure.

None of these were the case here.

**Comment by Nicholas-Westby** [comment]
I was curious if I could reproduce this on Umbraco 10.8.3 (the latest 10.x version), and I could reproduce it there as well. I thought I'd create a video of the tail end of the process this time so you can see that things are fine and then eventually fail:

https://github.com/umbraco/Umbraco-CMS/assets/5933222/7a11eec3-1d35-4d07-a47e-5b13c44bbbc2

I didn't bother with the ZIP for this one since the one for 13 and the instructions should suffice. Here are some differences with this iteration:

* Instead of 450ish widgets, this one has about 640 widgets per page (like the 13 version, each page has the same widgets).
* I got up to 35 pages with this one (the 13 one ran into issues at 7 pages).

I also got various errors on the homepage with the 10.8.3 one (even though I don't have any frontend rendering code, the doctype may have created one). The first error was the following (similar, but not the same, as the one from the video):

<img width="1025" alt="homepage-error" src="https://github.com/umbraco/Umbraco-CMS/assets/5933222/0907acd9-535d-409d-927e-00ca4358638b">

That was in the error logs (unlike the main error), so here's the stack trace for it:

```
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'length')
   at CSharpTest.Net.IO.TransactedCompoundFile.Write(UInt32 handle, Byte[] bytes, Int32 offset, Int32 length)
   at CSharpTest.Net.Storage.BTreeFileStoreV2.Update[T](IStorageHandle handleIn, ISerializer`1 serializer, T node)
   at CSharpTest.Net.Collections.LurchTable`2.TryDequeue(Predicate`1 predicate, KeyValuePair`2& value)
   at CSharpTest.Net.Collections.BPlusTree`2.StorageCache.Flush()
   at CSharpTest.Net.Collections.BPlusTree`2.StorageCache.Commit()
   at CSharpTest.Net.Collections.BPlusTree`2.CommitChanges(Boolean requiresLock)
   at CSharpTest.Net.Collections.BPlusTree`2..ctor(BPlusTreeOptions`2 ioptions)
   at Umbraco.Cms.Infrastructure.PublishedCache.DataSource.BTree.GetTree(String filepath, Boolean exists, NuCacheSettings settings, ContentDataSerializer contentDataSerializer)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.MainDomRegister()
   at Umbraco.Cms.Core.Runtime.MainDom.Register(Action install, Action release, Int32 weight)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.<EnsureCaches>b__52_0()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.EnsureCaches()
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.CreatePublishedSnapshot(String previewToken)
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at Umbraco.Cms.Web.Common.UmbracoContext.UmbracoContext.get_Content()
   at Umbraco.Cms.Web.Website.Routing.UmbracoRouteValueTransformer.TransformAsync(HttpContext httpContext, RouteValueDictionary values)
   at Microsoft.AspNetCore.Mvc.Routing.DynamicControllerEndpointMatcherPolicy.ApplyAsync(HttpContext httpContext, CandidateSet candidates)
   at Microsoft.AspNetCore.Routing.Matching.DfaMatcher.SelectEndpointWithPoliciesAsync(HttpContext httpContext, IEndpointSelectorPolicy[] policies, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|8_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
   at SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddleware.Invoke(HttpContext httpContext, Boolean retry)
   at StackExchange.Profiling.MiniProfilerMiddleware.Invoke(HttpContext context) in C:\projects\dotnet\src\MiniProfiler.AspNetCore\MiniProfilerMiddleware.cs:line 119
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Umbraco.Cms.Web.Common.Middleware.PreviewAuthenticationMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestLoggingMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

And here's another error from the logs that happened on the subsequent homepage refresh:

```
System.IO.IOException: The process cannot access the file 'C:\r\Umbraco10SpeedTest\umbraco\Data\TEMP\NuCache\NuCache.Content.db' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at CSharpTest.Net.IO.TransactedCompoundFile..ctor(Options options)
   at CSharpTest.Net.Storage.BTreeFileStoreV2..ctor(Options options)
   at CSharpTest.Net.Collections.BPlusTree`2.OptionsV2.CreateStorage()
   at CSharpTest.Net.Collections.BPlusTree`2.NodeCacheBase..ctor(BPlusTreeOptions`2 options)
   at CSharpTest.Net.Collections.BPlusTree`2.NodeCacheNone..ctor(BPlusTreeOptions`2 options)
   at CSharpTest.Net.Collections.BPlusTree`2..ctor(BPlusTreeOptions`2 ioptions)
   at Umbraco.Cms.Infrastructure.PublishedCache.DataSource.BTree.GetTree(String filepath, Boolean exists, NuCacheSettings settings, ContentDataSerializer contentDataSerializer)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.MainDomRegister()
   at Umbraco.Cms.Core.Runtime.MainDom.Register(Action install, Action release, Int32 weight)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.<EnsureCaches>b__52_0()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.EnsureCaches()
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.CreatePublishedSnapshot(String previewToken)
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at Umbraco.Cms.Web.Common.UmbracoContext.UmbracoContext.get_Content()
   at Umbraco.Cms.Web.Website.Routing.UmbracoRouteValueTransformer.TransformAsync(HttpContext httpContext, RouteValueDictionary values)
   at Microsoft.AspNetCore.Mvc.Routing.DynamicControllerEndpointMatcherPolicy.ApplyAsync(HttpContext httpContext, CandidateSet candidates)
   at Microsoft.AspNetCore.Routing.Matching.DfaMatcher.SelectEndpointWithPoliciesAsync(HttpContext httpContext, IEndpointSelectorPolicy[] policies, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|8_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
   at SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddleware.Invoke(HttpContext httpContext, Boolean retry)
   at StackExchange.Profiling.MiniProfilerMiddleware.Invoke(HttpContext context) in C:\projects\dotnet\src\MiniProfiler.AspNetCore\MiniProfilerMiddleware.cs:line 119
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Umbraco.Cms.Web.Common.Middleware.PreviewAuthenticationMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Umbraco.Cms.Web.Common.Middleware.UmbracoRequestLoggingMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

Maybe those will offer some additional clues (or they could be irrelevant, not sure).

One observation I can make is that this is apparently a longstanding bug, and it should be readily reproducible.

**Comment by Nicholas-Westby** [comment]
BTW, you may run into some other errors when trying to run the solution I sent you, even when just attempting to click a content node in the back office, such as this error:

```

Received an error from the server
An error occurred
Specified argument was out of the range of valid values. (Parameter 'length')

Exception Details
System.ArgumentOutOfRangeException, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e: Specified argument was out of the range of valid values. (Parameter 'length')
Stacktrace
at CSharpTest.Net.IO.TransactedCompoundFile.Write(UInt32 handle, Byte[] bytes, Int32 offset, Int32 length)
   at CSharpTest.Net.Storage.BTreeFileStoreV2.Update[T](IStorageHandle handleIn, ISerializer`1 serializer, T node)
   at CSharpTest.Net.Collections.BPlusTree`2.StorageCache.OnItemRemoved(KeyValuePair`2 item)
   at CSharpTest.Net.Collections.LurchTable`2.TryDequeue(Predicate`1 predicate, KeyValuePair`2& value)
   at CSharpTest.Net.Collections.LurchTable`2.TryDequeue(KeyValuePair`2& value)
   at CSharpTest.Net.Collections.BPlusTree`2.StorageCache.Flush()
   at CSharpTest.Net.Collections.BPlusTree`2.StorageCache.Commit()
   at CSharpTest.Net.Collections.BPlusTree`2.CommitChanges(Boolean requiresLock)
   at CSharpTest.Net.Collections.BPlusTree`2.BulkInsert(IEnumerable`1 items, BulkInsertOptions bulkOptions)
   at CSharpTest.Net.Collections.BPlusTree`2..ctor(BPlusTreeOptions`2 ioptions)
   at CSharpTest.Net.Collections.BPlusTree`2..ctor(OptionsV2 optionsV2)
   at Umbraco.Cms.Infrastructure.PublishedCache.DataSource.BTree.GetTree(String filepath, Boolean exists, NuCacheSettings settings, ContentDataSerializer contentDataSerializer)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.MainDomRegister()
   at Umbraco.Cms.Core.Runtime.MainDom.Register(Action install, Action release, Int32 weight)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.<EnsureCaches>b__55_0()
   at System.Threading.LazyInitializer.EnsureInitializedCore[T](T& target, Boolean& initialized, Object& syncLock, Func`1 valueFactory)
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.EnsureCaches()
   at Umbraco.Cms.Infrastructure.PublishedCache.PublishedSnapshotService.CreatePublishedSnapshot(String previewToken)
   at Umbraco.Cms.Web.Common.UmbracoContext.UmbracoContext.<>c__DisplayClass11_0.<.ctor>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location ---
   at System.Lazy`1.CreateValue()
   at Umbraco.Cms.Web.Common.UmbracoContext.UmbracoContext.get_PublishedSnapshot()
   at Umbraco.Cms.Web.Common.UmbracoContext.UmbracoContext.get_Content()
   at Umbraco.Cms.Core.Routing.AliasUrlProvider.GetOtherUrls(Int32 id, Uri current)+MoveNext()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.SparseArrayBuilder`1.ReserveOrAdd(IEnumerable`1 items)
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.ToArray()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.OrderedEnumerable`1.GetEnumerator()+MoveNext()
   at Umbraco.Extensions.UrlProviderExtensions.GetContentUrlsAsync(IContent content, IPublishedRouter publishedRouter, IUmbracoContext umbracoContext, ILocalizationService localizationService, ILocalizedTextService textService, IContentService contentService, IVariationContextAccessor variationContextAccessor, ILogger`1 logger, UriUtility uriUtility, IPublishedUrlProvider publishedUrlProvider)
   at Umbraco.Cms.Web.BackOffice.Mapping.ContentMapDefinition.GetUrls(IContent source)
   at Umbraco.Cms.Web.BackOffice.Mapping.ContentMapDefinition.Map[TVariant](IContent source, ContentItemDisplay`1 target, MapperContext context)
   at Umbraco.Cms.Core.Mapping.UmbracoMapper.<>c__DisplayClass11_0`2.<Define>b__1(Object source, Object target, MapperContext context)
   at Umbraco.Cms.Core.Mapping.UmbracoMapper.Map[TTarget](Object source, Type sourceType, MapperContext context)
   at Umbraco.Cms.Core.Mapping.UmbracoMapper.Map[TTarget](Object source, MapperContext context)
   at Umbraco.Cms.Core.Mapping.UmbracoMapper.Map[TTarget](Object source, Action`1 f)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.MapToDisplayWithSchedule(IContent content)
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentController.GetById(Int32 id)
   at lambda_method405(Closure, Object, Object[])
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextExceptionFilterAsync>g__Awaited|26_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
```

Here is a screenshot of that:

<img width="960" alt="view-node-error" src="https://github.com/umbraco/Umbraco-CMS/assets/5933222/60886211-9bb3-47db-9b53-5a00113eafc0">

If so, it might be a better option to recreate the issue from scratch (though fixing this issue and the various related ones would be worth tackling as part of this as well).

**Comment by Nicholas-Westby** [comment]
Here's a database backup of the Umbraco 13 site with just a single page that has 512 widgets:

[umbraco13speed.zip](https://github.com/umbraco/Umbraco-CMS/files/13826484/umbraco13speed.zip)

That should help to reproduce the issue with minimal fuss (so you can start there and being copying the page repeatedly). Clearing TEMP folders, restarting VS, rebuilding, and setting the start URL to `/umbraco/` may also help.

**Comment by Migaroez** [comment]
Hey @Nicholas-Westby 

This was a fun one to figure out. I could not reproduce it with my own setup but using your backup I was able to get the errors straight away.

So the issue is that BTree, a system we use in the NuCache, can not chop up the value of a property as we pass it in as 1 big thing. Blocklists/grids produce json that has all the data for all the configured blocks in 1 property in them.

With your setup, this results in a very big string value that can't be stored in the default blockSize. Luckily you can increase the size of the btree blocks pretty easily in the appsettings
https://docs.umbraco.com/umbraco-cms/reference/configuration/nucachesettings

As I understand it, changing this setting can have a performance impact as it is used for memory allocation and efficient memory block paging/prefetching. So sticking to a power of 2 for this setting is probably the best idea.

All the different errors point to the same origin: The Nucache being updated and thus having to write something to it's files.

The reason for the many `The following value was recognized as JSON but could not be parsed:` warnings is because when any block related property is saved, we check whether all required udi's are in a good state. Because some of the nested values in your properties also use json, we have to check wether this is json that is part of the block structure as blocks can be nested. When we detect valid json that is not in the format we expected, this error is shown. As there **might** be something wrong.

Please let me know if updating the appsettings works.

**Comment by Migaroez** [comment]
Since no feedback was supplied after the proposed solution, I will be closing this issue.

**Comment by Nicholas-Westby** [comment]
@Migaroez I still plan to get to this, just haven't had time yet.

**Comment by sotirisf** [comment]
I landed here having a similar problem with a 13.2.0 and I'm glad to say that changing the nucache settings worked. 

I wonder, though, if the BTreeBlockSize setting corresponds to the actual property size or chops up the data in some peculiar way, since I'm using a single BlockList with about 20 (large) blocks with sub-blocks inside them and, dumping the json directly from the database into a text file showed a total size 65k. However, it worked with a BTreeBlockSize of 16384.
